lib/Subscription/Manage.php

```diff
@@ -48,6 +54,19 @@ function onSave() {
     $this->url_helper->redirectBack();
   }
 
+  private function addHiddenSegments(Subscriber $subscriber, array $data) {
+    $shown_segments = $this->settings->get('subscription.segments');
+    if (empty($shown_segments)) return $data;
+    if (empty($data['segments'])) $data['segments'] = [];
```

Comment:
What does this line do? The `$data` variable must be an array therefore `empty($data)` already is `[]`.

Reply:
I am checking and initializing `$data['segments']` not `$data`